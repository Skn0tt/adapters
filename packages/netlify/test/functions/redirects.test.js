import { fileURLToPath } from 'url';
import { expect } from 'chai';
import fs from 'fs/promises';
import { cli } from './test-utils.js';

const root = new URL('../functions/fixtures/redirects/', import.meta.url).toString();

describe('SSG - Redirects', () => {
	before(async () => {
		await cli('build', '--root', fileURLToPath(root));
	});

	it('Creates a redirects file', async () => {
		let redirects = await fs.readFile(new URL('./dist/_redirects', root), 'utf-8');
		let parts = redirects.split(/\s+/);
		expect(parts).to.deep.equal([
			'',
			'/other',
			'/',
			'301',
			'',
		]);
		expect(redirects).to.matchSnapshot();
	});

	it('Does not create .html files', async () => {
		let hasErrored = false;
		try {
			await fixture.readFile('/other/index.html');
		} catch {
			hasErrored = true;
		}
		expect(hasErrored).to.equal(true, 'this file should not exist');
	});
});
