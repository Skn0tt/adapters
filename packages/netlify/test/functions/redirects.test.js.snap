// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSG - Redirects Creates a redirects file 1`] = `
"
/other    /       301
"
`;

exports[`SSR - Redirects Creates a redirects file 1`] = `
"
/other    /       301
"
`;
